Resources/public/js/views/ez-trashview.js

```diff
@@ -79,13 +87,69 @@ YUI.add('ez-trashview', function (Y) {
          * @return {Array} of JSONified `trashItems`
          */
         _convertTrashItemsToJSON: function () {
-            return Y.Array.map(this.get('trashItems'), function (trashItem) {
-                return {
-                    'item': trashItem.item.toJSON(),
-                    'parentLocation': trashItem.parentLocation.toJSON(),
-                    'contentType': trashItem.contentType.toJSON(),
-                };
+            var result = [];
+
+            Y.Object.each(this.get('trashItems'), function (value, key) {
+                result.push({
+                    'id': key,
+                    'item': value.item.toJSON(),
+                    'parentLocation': value.parentLocation.toJSON(),
+                    'contentType': value.contentType.toJSON(),
+                });
             });
+
+            return result;
+        },
+
+        /**
+         * Restores the selected trash items
+         *
+         * @private
+         * @method _restoreTrashItems
+         */
+        _restoreTrashItems: function (e) {
+            var selectedTrashItems = this.get('container').all('.ez-trashitem-box:checked'),
+                trashItems = [];
+
+            Y.each(selectedTrashItems, function (trashItem) {
```

Comment:
`selectedTrashItems` is a NodeList, to iterate on it you have [a `each` method](http://yuilibrary.com/yui/docs/api/classes/NodeList.html#method_each).